src/app-layout/visual-refresh/trigger-button.tsx

```diff
@@ -33,37 +52,69 @@ function TriggerButton(
     ariaControls,
     onClick,
     testId,
+    disabled = false,
     badge,
     selected = false,
     highContrastHeader,
   }: TriggerButtonProps,
   ref: React.Ref<ButtonProps.Ref>
 ) {
+  const { isMobile } = useAppLayoutInternals();
+
   return (
-    <div className={clsx(styles['trigger-wrapper'], !highContrastHeader && styles['remove-high-contrast-header'])}>
-      <button
-        aria-expanded={ariaExpanded}
-        aria-controls={ariaControls}
-        aria-haspopup={true}
-        aria-label={ariaLabel}
-        className={clsx(
-          styles.trigger,
-          {
-            [styles.selected]: selected,
-            [styles.badge]: badge,
-          },
-          className
-        )}
-        onClick={onClick}
-        ref={ref as React.Ref<HTMLButtonElement>}
-        type="button"
-        data-testid={testId}
-      >
-        <span className={clsx(badge && styles['trigger-badge-wrapper'])}>
-          <Icon name={iconName} svg={iconSvg} />
-        </span>
-      </button>
-      {badge && <div className={styles.dot} />}
+    <div
+      className={clsx(styles['trigger-wrapper'], !highContrastHeader && styles['remove-high-contrast-header'])}
+      data-testid={`${testId ? `${testId}-wrapper` : 'awsui-app-layout-trigger-wrapper'}`}
+    >
+      {isMobile ? (
```

Comment:
Why do we need to render the button differently for mobile view? The current implementation achieves this without any special conditions, so what's the purpose of making it more complicated?

Reply:
I'm under the assumption there is a reason for it, so I kept it https://github.com/cloudscape-design/components/blob/main/src/app-layout/visual-refresh/drawers.tsx#L342

Reply:
What does the drawers component from the link above do with this change? Also, what is the purpose of the isMobile condition?

Reply:
i removed the testid from here, if needed for the tooltip I will add there.  The isMobile is used to determine which component to render. There is a different layout on mobile. When the drawer is open it covers the toolbar

Reply:
If the layout is different on mobile, how is it currently being handled without this condition? What caught my attention is this sentence in the PR description:

> No change in functionality or UI

Could you clarify how the mobile layout is managed in the current implementation?

Reply:
I see. But you didn't replace the existing trigger button in the mobile triggers. Is that intentional?

Also, can we use InternalButton consistently and remove the implementation based on <button>?

Reply:
The tooltips will eventually be rendered as a sibiling element to the trigger button. The trigger buttons are intended to work as they do currently, but the logic for the tooltip and its handlers will be added here and rendered when called for based on props that will be added here.